hook/gomod/gomod.go

```diff
@@ -0,0 +1,92 @@
+package gomod
+
+import (
+	"path/filepath"
+
+	"golang.org/x/exp/maps"
+
+	"github.com/aquasecurity/fanal/hook"
+	"github.com/aquasecurity/fanal/types"
+)
+
+func init() {
+	hook.RegisterHook(gomodMergeHook{})
+}
+
+const version = 1
+
+type gomodMergeHook struct{}
+
+// Hook merges go.mod and go.sum.
+func (h gomodMergeHook) Hook(blob *types.BlobInfo) error {
+	var apps []types.Application
+	for _, app := range blob.Applications {
+		dir, file := filepath.Split(app.FilePath)
+		if app.Type == types.GoMod {
+			// go.sum should be merged to go.mod.
+			if file == "go.sum" {
+				continue
+			} else if file == "go.mod" && go116OrLess(app) {
```

Comment:
Does `go116OrLess` really need? 
`go-dep-parser` skips `inderect`, doesn't it?
https://github.com/aquasecurity/go-dep-parser/blob/d0501f906d906a5da17ae4ea2edc78a8c5fe062e/pkg/golang/mod/parse.go#L29

Reply:
That is exactly how we know it is Go 1.17+. If it is Go 1.17+, we should not merge go.sum.

Reply:
ok, I thought that `do-dep-parser` skips `inderect` dependencies. my mistake.